tests/cmdline/commands/test_process.py

```diff
@@ -123,6 +132,26 @@ def test_pause_play_kill(self):
         self.assertTrue(calc.is_terminated)
         self.assertTrue(calc.is_killed)
 
+    def test_pause_play_kill_nopk(self):
+        """
+        Test the pause/play/kill commands
+        """
+        # pylint: disable=no-member
+        calc = self.setup_waiting_process()
+
+        result = self.cli_runner.invoke(cmd_process.process_pause, ['--wait', str(calc.pk)])
+        self.assertIsNone(result.exception, result.output)
+        self.assertTrue(calc.paused)
+
+        result = self.cli_runner.invoke(cmd_process.process_play, ['--wait'])
+        self.assertIsNone(result.exception, result.output)
+        self.assertFalse(calc.paused)
+
+        result = self.cli_runner.invoke(cmd_process.process_kill, ['--wait', str(calc.pk)])
+        self.assertIsNone(result.exception, result.output)
+        self.assertTrue(calc.is_terminated)
+        self.assertTrue(calc.is_killed)
```

Comment:
Am I missing something or does this test not really pertain to the changes in the code? You are simply adding this as an additional test. Fine in principle, just to make sure.

On a side note, given the high degree of overlap with `test_pause_play_kill` these two would be a perfect candidate for `pytest.mark.parametrize`.

Reply:
Ah, sorry. Now I see I forgot to add the `-all` flag.

> On a side note, given the high degree of overlap with `test_pause_play_kill` these two would be a perfect candidate for `pytest.mark.parametrize`.

Mmm yeah, I agree, but this test is using the old testing structure. Can I use pytest fixtures on `AiidaTestCase` methods?

Reply:
no, but you can start the migration to using `pytest`. We have been doing this bit by bit whenever we touch tests that are in the old style. Would be good opportunity to do it for this file

Reply:
Ah, I see. Haha, that is some sneaky way to suggest to migrate the test :sweat_smile:...

But yeah, I gave it a shot. It was not easy because I have to try a couple of things to know what parts of the `setUp` and `tearDown` where necessary with the daemon being a relevant part of the test, but I think I manage to find a configuration that at least seems to work.

Ironically, in the end the test was not such a good candidate for `pytest.mark.parametrize` because the parameter actually relied on the pk of a node that has to be created during the test. Anyways I did parametrize using a boolean, but it was not as clean as originally expected.